### PR TITLE
add chart resize MWPW-112996

### PR DESCRIPTION
**Pull request by meganthecoder**
* Add dynamic chart resize
* Make size responsive
* Cleanup
* Add throttle util

(Will add tests next)

Resolves: [MWPW-112996](https://jira.corp.adobe.com/browse/MWPW-112996)

**Test URLs:**
- Before: https://data-viz--milo--adobecom.hlx.page/drafts/methomas/data-viz
- After: https://chart-resize--milo--adobecom.hlx.page/drafts/methomas/data-viz

### CODE REVIEW

File: libs/blocks/chart/chart.js [hunk 1]
```diff
@@ -178,51 +147,130 @@ const handleIntersect = (chart, chartOptions) => (entries, observer) => {
...
+  }
+
+  if (previousIsLarge !== currentIsLarge) {
+    chartInstance?.dispose();
+
+    const themeName = getTheme(currentSize);
+    const chart = window.echarts?.init(chartWrapper, themeName, { renderer: 'svg' });
+    const chartOptions = getChartOptions(chartType, data, colors, currentSize);
+
+    chart.setOption(chartOptions);
```
Comment (chrischrischris) [inline libs/blocks/chart/chart.js:232]: Do you need to setup the IntersectionObserver again?

File: libs/blocks/chart/chart.js [hunk 2]
```diff
@@ -178,51 +147,130 @@ const handleIntersect = (chart, chartOptions) => (entries, observer) => {
...
+      large: { minHeight: 350, width: '100%' },
+    },
+  };
+  let containerSize = containerSizes?.default?.[chartSize] || {};
+
+  if (chartType in containerSizes) {
+    containerSize = containerSizes?.[chartType]?.[chartSize] || {};
+  }
+
+  return containerSize;
```
[the remark this answers is not in the document]
Comment (chrischrischris) [reply libs/blocks/chart/chart.js:179]: This is super nit, i'm fine either way

**Review by JasonHowellSlavin — APPROVED** [review]
Good work! 

Looks like there are some good candidates for unit tests here in `getResponsiveSize` and `updateContainerSize`. Though you could get coverage for both in `handleResize` I think I remember you saying there were issues mocking the library.

File: libs/utils/utils.js [hunk 3]
```diff
@@ -329,3 +329,20 @@ export function isEmptyObject(obj) {
...
+    const time = new Date().getTime();
+    let timeout = null;
+
+    if (!previousCall || time - previousCall >= delay) {
+      previousCall = time;
+      throttled.apply(null, [opts, args]);
+      timeout = setTimeout(() => {
+        throttled.apply(null, [opts, args]);
+        timeout = null;
+      }, (delay * 2));
```
[the remark this answers is not in the document]
Comment (meganthecoder) [reply libs/utils/utils.js]: Yeah that's weird. I pulled this directly from dexterUI-tools. I've removed it now

File: libs/blocks/chart/chart.js (hunk 1 above)
Comment (meganthecoder) [reply libs/blocks/chart/chart.js:229]: I'll look into this and do some testing. If it needs to be updated, I'll create a new PR

**Pull request merged**